spacy/tests/lang/uk/test_lemmatizer.py

```diff
@@ -8,12 +8,22 @@
 def test_uk_lemmatizer(uk_lemmatizer):
     """Check that the default uk lemmatizer runs."""
     doc = Doc(uk_lemmatizer.vocab, words=["a", "b", "c"])
+    assert uk_lemmatizer.mode == "pymorphy3"
     uk_lemmatizer(doc)
     assert [token.lemma for token in doc]
 
 
-def test_uk_lookup_lemmatizer(uk_lookup_lemmatizer):
-    """Check that the lookup uk lemmatizer runs."""
-    doc = Doc(uk_lookup_lemmatizer.vocab, words=["a", "b", "c"])
-    uk_lookup_lemmatizer(doc)
-    assert [token.lemma for token in doc]
+@pytest.mark.parametrize(
+    "word,lemma",
+    (
+        ("якийсь", "якийсь"),
+        pytest.param("зеленої", "зелений", marks=pytest.mark.xfail()),
```

Comment:
Is there a specific reason for adding these failing tests?

Reply:
I included all the tests added in #11732, some of which had been commented out as failing in that PR and which had been added for pymorphy2. When I tested again with pymorphy3 there were a few minor differences, so it's not the exact same set as in #11732.

Reply:
@ richardpaulhudson This test is not failing in my PR. Did you try running while on the original PR branch?

Reply:
This is what I am getting with the test. I have only two commented out pieces and `"зеленої", "зелений"` is not one of them

```
rootdir: /Users/dlituiev/repos/spaCy, configfile: setup.cfg
plugins: hypothesis-6.56.2
collected 2 items

spacy/tests/pipeline/test_pymorphy_lemmatizer.py ..                                                     [100%]

============================================== warnings summary ===============================================
spacy/tests/pipeline/test_pymorphy_lemmatizer.py: 23 warnings
  /opt/anaconda3/envs/spacy_dev/lib/python3.8/site-packages/pymorphy2/units/base.py:70: DeprecationWarning: inspect.getargspec() is deprecated since Python 3.0, use inspect.signature() or inspect.getfullargspec()
    args, varargs, kw, default = inspect.getargspec(cls.__init__)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
======================================= 2 passed, 23 warnings in 0.24s ========================================
```


```
def test_lookup_lemmatizer_uk():
    nlp = Ukrainian()
    lemmatizer = nlp.add_pipe("lemmatizer", config={"mode": "pymorphy2_lookup"})
    assert isinstance(lemmatizer.lookups, Lookups)
    assert not lemmatizer.lookups.tables
    assert lemmatizer.mode == "pymorphy2_lookup"
    nlp.initialize()
    assert nlp("якась")[0].lemma_ == "якийсь"
    assert nlp("якийсь")[0].lemma_ == "якийсь"
    assert nlp("зеленої")[0].lemma_ == 'зелений'
    assert nlp("розповідають")[0].lemma_ == 'розповідати'
    assert nlp("розповіси")[0].lemma_ == 'розповісти'
    # assert nlp("телятові")[0].lemma_ == 'теля' # pymorph2 fails

def test_lookup_lemmatizer_ru():
    nlp = Russian()
    lemmatizer = nlp.add_pipe("lemmatizer", config={"mode": "pymorphy2_lookup"})
    assert isinstance(lemmatizer.lookups, Lookups)
    assert not lemmatizer.lookups.tables
    assert lemmatizer.mode == "pymorphy2_lookup"
    nlp.initialize()
    assert nlp("бременем")[0].lemma_ == 'бремя'
    assert nlp("будешь")[0].lemma_ == "быть"
    # assert nlp("какая-то")[0].lemma_ == "какой-то" # fails due to faulty word splitting
    assert nlp("зелёной")[0].lemma_ == 'зелёный'
```

Reply:
I think your tests were using pymorphy2? With pymorphy3 it fails on my end with:

```
>       assert uk_lookup_lemmatizer(doc)[0].lemma_ == lemma
E       AssertionError: assert 'зеленої' == 'зелений'
```

Reply:
Yes, it works for `pymorphy2` and fails for `pymorphy3`. I will investigate it. Might be an issue with `pymorphy3` itself

Reply:
The test file goes:
` lemmatizer = nlp.add_pipe("lemmatizer", config={"mode": "pymorphy2_lookup"})`

so yes, pymorphy2

Reply:
pymorphy3 treats it as a surname also (disregarding small case) and for some reason assigns initial form as `зелена`, though masculine form exist. So the issue is with pymorphy3.

```
[Parse(word='зеленої', tag=OpencorporaTag('NOUN,Surn,femn,anim gent'), normal_form='зелена', score=1.0, methods_stack=((DictionaryAnalyzer(), 'зеленої', 27, 1),)), Parse(word='зеленої', tag=OpencorporaTag('ADJF femn,gent,compb'), normal_form='зелений', score=1.0, methods_stack=((DictionaryAnalyzer(), 'зеленої', 2674, 11),))]
```

Reply:
There seems to be a broader issue with Pymorphy3 treating adjectives as surnames, e.g. `чорної` similarly gives

```
[Parse(word='чорної', tag=OpencorporaTag('NOUN,Surn,femn,anim gent'), normal_form='чорна', score=1.0, methods_stack=((DictionaryAnalyzer(), 'чорної', 27, 1),)), Parse(word='чорної', tag=OpencorporaTag('NOUN,femn,inan gent'), normal_form='чорна', score=1.0, methods_stack=((DictionaryAnalyzer(), 'чорної', 356, 1),)), Parse(word='чорної', tag=OpencorporaTag('ADJF femn,gent,compb'), normal_form='чорний', score=1.0, methods_stack=((DictionaryAnalyzer(), 'чорної', 2425, 11),)), Parse(word='чорної', tag=OpencorporaTag('NOUN,anim femn,gent'), normal_form='чорний', score=1.0, methods_stack=((DictionaryAnalyzer(), 'чорної', 4456, 9),))]
```

As any adjective theoretically can be a surname (while behaving morphologically as an adjective anyways), this throws a wrench into rule-based lemmatization. Unfortunately pymorphy3 has no issue tracking as of today. I've reached out to the maintainer directly. I suggest to keep the test for `pymorphy2` and comment it out for `pymorphy3` for now